src/librustc_resolve/lib.rs

```diff
@@ -1845,6 +1844,7 @@ impl<'a> Resolver<'a> {
 
                         for trait_item in trait_items {
                             this.check_proc_macro_attrs(&trait_item.attrs);
+                            this.visit_generics(&trait_item.generics);
```

Comment:
we should not be invoking `visit_generics` here. IT will happen when we invoke `walk_trait_item` below. Instead, we should be inserting the type-parameters rib:

```rust
let type_parameters = HasTypeParameters(&trait_item.generics, MethodRibKind(!sig.decl.has_self()));
this.with_type_parameter_rib(type_parameters, |this| { /* old code goes here */ })
```